### PR TITLE
FMWK-54 Implement find by POJO field using comparison operators

### PR DESCRIPTION
**Pull request by agrgr**
Currently without filter flow (secondary index)

### CODE REVIEW

File: src/main/java/org/springframework/data/aerospike/query/Qualifier.java [hunk 1]
```diff
@@ -324,6 +337,73 @@ public Exp toFilterExp() {
...
+				} else {
+					exp = Exp.eq(
+							MapExp.getByKey(MapReturnType.VALUE, Exp.Type.INT, Exp.val(getValue2().toString()), Exp.mapBin(getField())),
+							Exp.val(getValue1().toLong()));
+				}
+				break;
+			case MAP_KEY_VALUE_GT:
+				// VALUE2 contains key (field name)
+				if (getValue1().getType() == ParticleType.STRING) {
+					throw new AerospikeException("FilterExpression unsupported operation: " + getOperation() + "(String)");
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/query/Qualifier.java:355]: nit: duplicate Exception message

**Review by reugn — COMMENTED** [review]
Update the Javadoc of the Qualifier class with supported operations.

File: src/main/java/org/springframework/data/aerospike/query/Qualifier.java [hunk 2]
```diff
@@ -127,6 +128,8 @@ public static String getStringEquals(String base) {
 	public enum FilterOperation {
 		EQ, GT, GTEQ, LT, LTEQ, NOTEQ, BETWEEN, START_WITH, ENDS_WITH, CONTAINING, IN,
 		LIST_CONTAINS, MAP_KEYS_CONTAINS, MAP_VALUES_CONTAINS,
+		MAP_KEY_VALUE_EQ, MAP_KEY_VALUE_GT, MAP_KEY_VALUE_GTEQ,
+		MAP_KEY_VALUE_LT, MAP_KEY_VALUE_LTEQ, MAP_KEY_VALUE_BETWEEN,
```
Comment (roimenashe) [inline src/main/java/org/springframework/data/aerospike/query/Qualifier.java:132]: Can we also support `NOTEQ`, `START_WITH`, `ENDS_WITH` and `IN`? (`IN` is a special case that requires more investigation), I don't see a reason not to for the other ones.

Comment (agrgr) [reply src/main/java/org/springframework/data/aerospike/query/Qualifier.java:170]: Added NOTEQ, START_WITH, ENDS_WITH, CONTAINING

File: src/main/java/org/springframework/data/aerospike/query/Qualifier.java (hunk 1 above)
Comment (agrgr) [reply src/main/java/org/springframework/data/aerospike/query/Qualifier.java]: Updated exceptions, added more of them. Perhaps will add a common "checkCorrespondingType" method / factory method for exceptions

Comment (agrgr) [reply src/main/java/org/springframework/data/aerospike/query/Qualifier.java]: Updated Javadoc

File: src/main/java/org/springframework/data/aerospike/query/Qualifier.java [hunk 3]
```diff
@@ -36,33 +36,25 @@
...
-	private static final String mapIterVar = "mapIterVar";
-	private static final String FIELD = "field";
-	private static final String IGNORE_CASE = "ignoreCase";
-	private static final String VALUE2 = "value2";
-	private static final String VALUE1 = "value1";
-	private static final String QUALIFIERS = "qualifiers";
-	private static final String OPERATION = "operation";
-	private static final String AS_FILTER = "queryAsFilter";
+//	private static final String listIterVar = "listIterVar";
+//	private static final String mapIterVar = "mapIterVar";
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/query/Qualifier.java:49]: Need to be removed?

**Review by reugn — APPROVED** [review]

File: src/main/java/org/springframework/data/aerospike/query/Qualifier.java (hunk 3 above)
Comment (agrgr) [reply src/main/java/org/springframework/data/aerospike/query/Qualifier.java:49]: So we can remove it, I guess.

**Review by roimenashe — APPROVED** [review]

**Pull request merged**